src/main/java/sirius/biz/util/ExtractedZipFile.java

```diff
@@ -25,18 +24,18 @@
 class ExtractedZipFile implements ExtractedFile {
 
     private final ZipEntry entry;
-    private final InputStream zipInputStream;
+    private final Supplier<InputStream> inputStreamSupplier;
     private final Amount progress;
 
-    ExtractedZipFile(ZipEntry entry, InputStream zipInputStream, Amount progress) {
+    ExtractedZipFile(ZipEntry entry, Supplier<InputStream> inputStreamSupplier, Amount progress) {
         this.entry = entry;
-        this.zipInputStream = zipInputStream;
+        this.inputStreamSupplier = inputStreamSupplier;
         this.progress = progress;
     }
 
     @Override
     public InputStream openInputStream() throws IOException {
-        return new CloseShieldInputStream(zipInputStream);
+        return new CloseShieldInputStream(inputStreamSupplier.get());
```

Comment:
catch and rethrow IOException catch and properly handle all other exceptions as .handled().